MLOps/intro-to-ml-model-deployment-and-management/intro-to-ml-model-deployment-and-management.md

```diff
@@ -0,0 +1,1010 @@
+num_pumps = 1
+
+X_pd = X.to_pandas()
+
+for i in range(num_pumps):
+    for row in range(X.nrows):
+        print(list(X.names))
+        print(list(list(X_pd.iloc[row].values)))
+        response = requests.post(
+            url = scoring_endpoint,
+            json = {"fields": list(X.names), "rows": [[str(num) for num in list(X_pd.iloc[0].values)]]},
+        )
+        time.sleep(1)
+
+return X  # return dt.Frame, pd.DataFrame, np.ndarray or a list or named dict of those
+~~~
+
+![dai-recipe-live-code-data-pumper](./assets/dai-recipe-live-code-data-pumper.jpg)
+
+Alternatively, you could run a Python script from your CLI to execute the data pumper. Refer to the script file: [data-pumper-called-from-cli.py](./recipes/data-pumper-called-from-cli.py)
+
+9\. Click **save** to execute the data pumper
+
+10\. Go back to MLOps tab and go to the **hydraulic system** project page
+
+## Task 6: Monitor MLOps Deployed Model
+
+1\. In the **Deployments** table on the hydraulic system page, on the **hydraulic_system_2** row, click on **3 dots** actions and then click on **Monitoring**.
+
+The grafana dashboard with all the monitoring visualizations of our model deployment will be shown. Those visualizations will be updated every 30s by default, but you can adjust this number if you want the visualizations updated faster. For instance, this number has been updated to 10s.
+
+You can see metadata for alerts and column types:
+
+![mlops-deployed-model-monitoring-1](./assets/mlops-deployed-model-monitoring-1.jpg)
+
+2\. Scroll down the monitoring dashboard to see all the visualizations. 
+
+You can see a visualization for the deployed model's scoring latency:
+
+![mlops-deployed-model-monitoring-2](./assets/mlops-deployed-model-monitoring-2.jpg)
+
+You can see a scatter plot, histogram and heatmap visualization for psa_bar:
+
+![mlops-deployed-model-monitoring-3](./assets/mlops-deployed-model-monitoring-3.jpg)
+
+![mlops-deployed-model-monitoring-4](./assets/mlops-deployed-model-monitoring-4.jpg)
+
+As you scroll down the visualizations, you will see the samples collected increasing and visualizations for each column in the test set.
+
+3\. Keep scrolling until you get to the target column, which is **cool_cond_y** in our case:
+
+4\. We have a visualizations for cool_cond_y.3, cool_cond_y.20 and cool_cond_y.100. Let's look at cool_cond_y.3:
+
+![mlops-deployed-model-monitoring-5](./assets/mlops-deployed-model-monitoring-5.jpg)
+
+![mlops-deployed-model-monitoring-6](./assets/mlops-deployed-model-monitoring-6.jpg)
+
+What is happening with the variables used when making predictions? How are the visualization graphs for the target column?
+
+## Task 7: Challenge
+
+### Deploy New Driverless AI Model using MLOps
+
+You just learned how to deploy a Driverless AI model that predicts hydraulic cooling condition. You may have some other problems you are working on solving. Try deploying a new Driverless AI model from a different Diverless AI project using MLOps.
```

Comment:
Ooooh, this is a good challenge!  👍  An issue I often see with tutorials online is they are written to be `followed once`, and not necessarily act as a reference for regular use.  Adding this challenge seems like a good way to measure if the tutorial helped a user become self-sufficient when browsing around the product.

Reply:
@orendain thanks. It would be good to be able to measure if the tutorial helped users become self-sufficient.